includes/class-wcpdf-admin.php

```diff
@@ -790,150 +793,48 @@ public function get_current_values_for_document( $document, $data ) {
-							</div>
-						<?php endif; ?>										
-						<?php do_action( 'wpo_wcpdf_meta_box_after_document_data', $document, $document->order ); ?>
-					<?php else : ?>
-						<?php /* translators: document title */ ?>
-						<?php
-						if ( $this->user_can_manage_document( $document->get_type() ) ) {
-							printf(
-								'<span class="wpo-wcpdf-set-date-number button">%s</span>',
-								sprintf(
-									/* translators: document title */
-									esc_html__( 'Set %s number & date', 'woocommerce-pdf-invoices-packing-slips' ),
-									wp_kses_post( $document->get_title() )
-								)
-							); 
-						} else {
-							printf( '<p>%s</p>', esc_html__( 'You do not have sufficient permissions to edit this document.', 'woocommerce-pdf-invoices-packing-slips' ) );
-						}
-						?>
-						
-					<?php endif; ?>
-				</div>
-
-				<!-- Editable -->
-				<div class="editable">
-					<?php if( isset( $data['number'] ) ) : ?>
-					<p class="form-field <?= esc_attr( $data['number']['name'] ); ?>_field">	
-						<label for="<?= esc_attr( $data['number']['name'] ); ?>"><?= wp_kses_post( $data['number']['label'] ); ?></label>
-						<input type="text" class="short" style="" name="<?= esc_attr( $data['number']['name'] ); ?>" id="<?= esc_attr( $data['number']['name'] ); ?>" value="<?= esc_attr( $data['number']['plain'] ); ?>" disabled="disabled" > (<?= esc_html__( 'unformatted!', 'woocommerce-pdf-invoices-packing-slips' ); ?>)
-					</p>
-					<?php endif; ?>
-					<?php if( isset( $data['date'] ) ) : ?>
-					<p class="form-field form-field-wide">
-						<label for="<?= esc_attr( $data['date']['name'] ); ?>[date]"><?= wp_kses_post( $data['date']['label'] ); ?></label>
-						<input type="text" class="date-picker-field" name="<?= esc_attr( $data['date']['name'] ); ?>[date]" id="<?= esc_attr( $data['date']['name'] ); ?>[date]" maxlength="10" value="<?= esc_attr( $data['date']['date'] ); ?>" pattern="[0-9]{4}-(0[1-9]|1[012])-(0[1-9]|1[0-9]|2[0-9]|3[01])" disabled="disabled"/>@<input type="number" class="hour" disabled="disabled" placeholder="<?php esc_attr_e( 'h', 'woocommerce' ); ?>" name="<?= esc_attr( $data['date']['name'] ); ?>[hour]" id="<?= esc_attr( $data['date']['name'] ); ?>[hour]" min="0" max="23" size="2" value="<?= esc_attr( $data['date']['hour'] ); ?>" pattern="([01]?[0-9]{1}|2[0-3]{1})" />:<input type="number" class="minute" placeholder="<?php esc_attr_e( 'm', 'woocommerce' ); ?>" name="<?= esc_attr( $data['date']['name'] ); ?>[minute]" id="<?= esc_attr( $data['date']['name'] ); ?>[minute]" min="0" max="59" size="2" value="<?= esc_attr( $data['date']['minute'] ); ?>" pattern="[0-5]{1}[0-9]{1}"  disabled="disabled" />
-					</p>
-					<?php endif; ?>
-				</div>
 
-				<!-- Document Notes -->
-				<?php if( array_key_exists( 'notes', $data ) ) : ?>
+		$independent_documents = apply_filters( 'wpo_wcpdf_document_data_meta_box_independent_documents', array( 'invoice', 'packing-slip' ) );
```

Comment:
I think this could be problematic when you have other documents being generated on the background, which hides them from the document data while are being processing.

Reply:
Thanks for catching that. This should fix the issue: https://github.com/wpovernight/woocommerce-pdf-invoices-packing-slips/pull/757/commits/ab89e8c5e4d56b333c7b8849b6d3c582368319e0.